pennylane/ops/op_math/controlled.py

```diff
@@ -199,11 +199,29 @@ def create_controlled_op(op, control, control_values=None, work_wires=None):
     return _ctrl_transform(op, control, control_values, work_wires)
 
 
+def remove_from_queue_args_and_kwargs(item):
```

Comment:
Maybe move this helper method to the `tape` module and just call it something like `recursively_remove_operators_from_queue`.

Reply:
Sounds good, is okey in `tape/operation_recorder.py`?

Reply:
Might be more suitable in `pennylane/queueing.py` actually.

Reply:
We can use pytrees and simply do:
```
leaves, _ = qml.pytrees.flatten((args, kwargs), lambda obj: isinstance(obj, Operator))
_ = [qml.QueuingManager.remove(l) for l in leaves if isinstance(l, Operator)]
```

Given this is just two lines, I'd be worried that adding a public function adds more complexity and coupling that it solves. Duplicate code is not a bad thing when it helps keep modules simple and loosely coupled.

I'd prefer to just copy those two lines into `adjoint`, `controlled`, and `TransformDispatcher._qfunc_transform`.